mmv1/products/workstations/WorkstationConfig.yaml

```diff
@@ -179,96 +196,106 @@ properties:
           - !ruby/object:Api::Type::Array
             name: "tags"
             item_type: Api::Type::String
-            immutable: true
             description: |
               Network tags to add to the Compute Engine machines backing the Workstations.
           - !ruby/object:Api::Type::Boolean
             name: "disablePublicIpAddresses"
-            immutable: true
             description: |
               Whether instances have no public IP address.
           - !ruby/object:Api::Type::NestedObject
             name: "shieldedInstanceConfig"
             description: |
               A set of Compute Engine Shielded instance options.
-            immutable: true
+            default_from_api: true
             properties:
               - !ruby/object:Api::Type::Boolean
                 name: "enableSecureBoot"
-                immutable: true
                 description: |
                   Whether the instance has Secure Boot enabled.
               - !ruby/object:Api::Type::Boolean
                 name: "enableVtpm"
-                immutable: true
                 description: |
                   Whether the instance has the vTPM enabled.
               - !ruby/object:Api::Type::Boolean
                 name: "enableIntegrityMonitoring"
-                immutable: true
                 description: |
                   Whether the instance has integrity monitoring enabled.
           - !ruby/object:Api::Type::NestedObject
             name: "confidentialInstanceConfig"
             description: |
               A set of Compute Engine Confidential VM instance options.
-            immutable: true
+            default_from_api: true
             properties:
               - !ruby/object:Api::Type::Boolean
                 name: "enableConfidentialCompute"
-                immutable: true
+                # TODO(esu): Change this to required in next breaking release.
                 description: |
                   Whether the instance has confidential compute enabled.
   - !ruby/object:Api::Type::Array
     name: "persistentDirectories"
     description: |
       Directories to persist across workstation sessions.
     default_from_api: true
+    immutable: true
```

Comment:
Looks like we're changing some of the fields to be immutable. What's the current behavior if users try to update these fields? Will the API error out or just ignore the request silently?

Reply:
The current behavior is for the API to reject the requests (`persistentDirectories` has the same issue above as `host`, where the top-level key is never updatable).

At the time being `persistentDirectories` is guaranteed to be size 1, with the only updatable field being `persistentDirectories[0].gcePd.reclaimPolicy`. However, since it seems difficult to generate an indexed update mask, and it's not known if the array will always be this size, we opted to make all of the fields immutable for now.

Does that seem like a reasonable approach? (would it even be possible to generate a more specific update mask with a specific index?)

Reply:
After discussing with team, this field may end up restructured in the future, since it's not following update mask best practices. For the time being, we are considering all of the fields and subfields immutable.